src/Uniphar.Dotnet.Cli/Commands/GetTokenCommand.cs

```diff
@@ -11,25 +12,51 @@ public sealed class Settings : CommandSettings
         public required Guid ClientId { get; set; }
 
         [CommandOption("-s|--scope")]
-        [DefaultValue(".default")]
+        [DefaultValue(null)]
         public string? Scope { get; set; }
+
+        [CommandOption("--authority")]
+        [DefaultValue("ms")]
+        public string? Authority { get; set; }
     }
 
 
     public async override Task<int> ExecuteAsync(CommandContext context, Settings settings)
     {
-        var app = PublicClientApplicationBuilder.Create(settings.ClientId.ToString())
-                                               .WithTenantId(settings.TenantId.ToString())
-                                               .WithDefaultRedirectUri()
-                                               .Build();
-        
-        var scope = string.IsNullOrWhiteSpace(settings.Scope) ? ".default" : settings.Scope;
-
-        var result = await app.AcquireTokenInteractive([$"{settings.ClientId}/{scope}"]).ExecuteAsync();
-        
-        AnsiConsole.MarkupLine($"{Environment.NewLine}[purple]{result.AccessToken}[/]");
+        var tenantId = settings.TenantId.ToString();
+        var clientId = settings.ClientId.ToString();
+        var authority = settings.Authority ?? "ms";
+
+        var authorityUri = authority.ToLower() switch
+        {
+            "ms" => $"https://login.microsoftonline.com/{tenantId}",
+            "ciam" => $"https://{tenantId}.ciamlogin.com",
+            _ when Uri.TryCreate(authority, UriKind.Absolute, out _) => ((Func<string>)(() => {
```

Comment:
This is a bit syntactically busy, I'd suggest that we can make this a bit nicer by using a tuple and deconstruction:

```
var (authorityUri, isPredefinedAuthority) = authority.ToLower() switch
{
    "ms" => ($"https://login.microsoftonline.com/{tenantId}", true),
    "ciam" => ($"https://{tenantId}.ciamlogin.com", true),
    _ when Uri.TryCreate(authority, UriKind.Absolute, out _) => (authority, false),
    _ => throw new Exception("Invalid authority URI")
};

if (!isPredefinedAuthority)
	AnsiConsole.MarkupLine("[bold yellow]Warning:[/] Using an arbitrary URI for authority is temporary. Please submit a PR to add your authority (currently supported: ms, ciam).");

```